compiler/rustc_hir_typeck/src/pat.rs

```diff
@@ -758,6 +758,28 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
                         err.span_note(sp, format!("{msg}: `{sugg}`"));
                     }
                 }
+                hir::Node::Pat(pt) if let PatKind::TupleStruct(_, pat_arr, _) = pt.kind => {
+                    for i in pat_arr.iter() {
+                        if let PatKind::Ref(the_ref, _) = i.kind {
+                            if let PatKind::Binding(mt, _, ident, _) = the_ref.kind {
+                                let hir::BindingAnnotation(_, mtblty) = mt;
+                                let ss = match mtblty {
```

Comment:
I think you can just use https://doc.rust-lang.org/nightly/nightly-rustc/rustc_ast/ast/enum.Mutability.html#method.ref_prefix_str here